Healthy/Classes/Extensions/UIButton+Style.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  UIButton+Style.swift
+//  Healthy
+//
+//  Created by Ahmed Nasr on 13/05/2023.
+//
+
+import UIKit
+
+
+//MARK: - button style -
+//
+extension UIButton {
+    enum ButtonStyle {
+        case primary
+        case secondary
+        case disable
+    }
+}
+
+
+//MARK: - apply button style -
+//
+extension UIButton {
+    func applyButtonStyle(style: ButtonStyle) {
+        buttonStyleAttributes(style: style)
+    }
+}
+
+
+//MARK: - button style attribute -
+//
+extension UIButton {
+    private func buttonStyleAttributes(style: ButtonStyle) {
+        switch style {
+        case .primary:
+            backgroundColor = .black
+        case .secondary:
+            backgroundColor = .green
+        case .disable:
+            backgroundColor = .gray
+        }
+        
+        tintColor = .white
+        titleLabel?.font = UIFont.systemFont(ofSize: 32)
+        layer.cornerRadius = 12
+        layer.masksToBounds = true
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(equalToConstant: 40),
```

Comment:
- Same as above. 
- Also, this constraint should have high priority not required (default)

Reply:
@ahmdmhasn  Why should it have high priority not required? and what is the default? 1000?

Reply:
- Constants must have any attributes or just numbers.
- how to set high priority to button using code ???

Reply:
Default is required (1000) ... High has priority of (750)

Reply:
```swift
let widthConstraint = heightAnchor.constraint(equalToConstant: 40) 
widthConstraint.priority = .high 
widthConstraint.isActive = true
```